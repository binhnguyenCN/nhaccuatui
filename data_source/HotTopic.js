const HotTopicData = {
  title: {
    content: 'chủ đề hot',
    url: '/',
    icon: 'more',
  },
};
export default HotTopicData;
