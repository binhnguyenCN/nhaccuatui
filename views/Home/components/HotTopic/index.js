import React from 'react';

import styles from './styles.module.scss';

const HotTopic = () => (
  <div className={styles['hot-topic']}>
    <h1>HOT TOPIC</h1>
  </div>
);

export default HotTopic;
