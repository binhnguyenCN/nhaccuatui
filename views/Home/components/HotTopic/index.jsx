import React from 'react';

import styles from './styles.module.scss';

const HotTopic = () => {
  console.log({});
  return (
    <div className={styles['hot-topic']}>
      <h1>HOT TOPIC</h1>
    </div>
  );
};

export default HotTopic;
