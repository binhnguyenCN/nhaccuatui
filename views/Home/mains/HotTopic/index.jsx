// libs
import React from 'react';
// layout
import CategoryTitles from '../../../../data_source/CategoryTitle';
import CategoryTitle from '../../components/CategoryTitle';
import TopicList from './components/TopicList';
// others
import styles from './styles.module.scss';

const HotTopic = ({ topics }) => (
  <div className={styles['hot-topic-wrapper']}>
    <div className={styles['hot-topic-inner']}>
      <CategoryTitle title={CategoryTitles.hotTopic} />
      <TopicList topics={topics} />
    </div>
  </div>
);

export default HotTopic;
