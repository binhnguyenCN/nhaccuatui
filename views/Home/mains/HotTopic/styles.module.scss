.hot-topic {
  grid-area: hotTopic;
  background-color: red;
}
