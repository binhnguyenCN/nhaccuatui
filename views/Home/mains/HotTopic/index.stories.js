import React from 'react';
import { hotTopic } from '../../../../api/mockApi';

import HotTopic from './index';

export default {
  title: 'Pages/Home/main/HotTopic',
  component: HotTopic,
  parameters: { topics: hotTopic },
};
export const Template = () => <HotTopic topics={hotTopic} />;
