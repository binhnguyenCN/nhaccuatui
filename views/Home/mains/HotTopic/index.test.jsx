// libs
import renderer from 'react-test-renderer';
// layouts
import HotTopic from '.';
// mocks
import mockData from '@/mocks/mockApi';

describe('render hot topic', () => {
  it('should renders correctly', () => {
    // act
    const tree = renderer
      .create(<HotTopic topics={mockData.hotTopic} />)
      .toJSON();
    // assert
    expect(tree).toMatchSnapshot();
  });
});
