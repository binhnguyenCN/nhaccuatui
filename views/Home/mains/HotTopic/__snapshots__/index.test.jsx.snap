// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render hot topic should renders correctly 1`] = `
<div
  className="hot-topic-wrapper"
>
  <div
    className="hot-topic-inner"
  >
    <h3
      className="category-title"
    >
      <span
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span>
          chủ đề hot
        </span>
        <span>
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </span>
    </h3>
    <ul
      className="topic-list"
    >
      <li
        className="topic-item"
      >
        <span
          href="https://avatar-ex-swe.nixcdn.com/topic/image-1440/2020/09/04/e/3/7/5/1599203331252_640.png"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="topic"
            height={100}
            src="https://avatar-ex-swe.nixcdn.com/topic/mobile/2020/09/04/5/9/2/a/1599203330765.jpg"
            width={300}
          />
        </span>
      </li>
      <li
        className="topic-item"
      >
        <span
          href="https://avatar-ex-swe.nixcdn.com/topic/image-1440/2020/06/19/e/d/d/3/1592557329285_640.png"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="topic"
            height={100}
            src="https://avatar-ex-swe.nixcdn.com/topic/mobile/2020/06/17/7/b/3/b/1592361925321.jpg"
            width={300}
          />
        </span>
      </li>
      <li
        className="topic-item"
      >
        <span
          href="https://avatar-ex-swe.nixcdn.com/topic/image-1440/2020/08/13/0/0/b/a/1597291431177_640.png"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="topic"
            height={100}
            src="https://avatar-ex-swe.nixcdn.com/topic/mobile/2020/08/13/e/5/a/b/1597291430551.jpg"
            width={300}
          />
        </span>
      </li>
      <li
        className="topic-item"
      >
        <span
          href="https://avatar-ex-swe.nixcdn.com/topic/image-1440/2020/08/13/0/0/b/a/1597291766974_640.png"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="topic"
            height={100}
            src="https://avatar-ex-swe.nixcdn.com/topic/mobile/2020/08/13/e/5/a/b/1597291766403.jpg"
            width={300}
          />
        </span>
      </li>
      <li
        className="topic-item"
      >
        <span
          href="https://avatar-ex-swe.nixcdn.com/topic/image-1440/2021/03/11/4/6/a/e/1615451482817_640.png"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="topic"
            height={100}
            src="https://avatar-ex-swe.nixcdn.com/topic/mobile/2021/03/11/c/1/2/6/1615451437099.jpg"
            width={300}
          />
        </span>
      </li>
    </ul>
  </div>
</div>
`;
